cvprac/cvp_api.py

```diff
@@ -361,6 +362,86 @@ def get_inventory(self, start=0, end=0, query=''):
                     dev['containerName'] = ''
             return data
 
+    def add_devices_to_inventory(self, device_list):
+        ''' Add a list of devices to the specified parent container.
+
+            Args:
+                device_list (list): A list of devices to be added in the
+                    form of dictionaries. Each device dictionary should
+                    contain the following information:
+                    - device_ip (str): ip address of device we are adding
+                    - parent_name (str): Parent container name
+                    - parent_key (str): Parent container key
+
+                    Example:
+                    device_list = [
+                        {
+                            device_ip: '10.10.10.1',
+                            parent_name: 'Tenant',
+                            parent_key: 'root'
+                        },
+                        {
+                            device_ip: '10.10.10.2',
+                            parent_name: 'MyContainer',
+                            parent_key: 'container-id-1234'
+                        }
+                    ]
+        '''
+
+        self.log.debug('add_device_to_inventory: called')
+        if self.clnt.apiversion is None:
+            self.get_cvp_info()
+        if self.clnt.apiversion == 'v1':
+            self.log.debug('v1 Inventory API Call')
+            data_list = []
+            for device in device_list:
+                dev_data = {
+                    'containerName': device['parent_name'],
+                    'containerId': device['parent_key'],
+                    'containerType': 'Existing',
+                    'ipAddress': device['device_ip'],
+                    'containerList': []
+                }
+                data_list.append(dev_data)
+            data = {'data': data_list}
+            self.clnt.post('/inventory/add/addToInventory.do?'
+                           'startIndex=0&endIndex=0', data=data,
+                           timeout=self.request_timeout)
+        else:
+            self.log.debug('v2 Inventory API Call')
+
+            # First add the devices to the inventory in a single call
+            data = {'hosts': [x['device_ip'] for x in device_list]}
```

Comment:
Consolidate list creation of device_ips into one line up above this and remove 420.